### PR TITLE
bug fix of issue 29 when regproxy with an existing voters record

### LINKED ISSUE #29: regproxy results in faulty record in voters table if account already has a voteproducer record
if you regproxy on an account that already has a regular voteproducer record in voters table, then the result is that neither the fioaddress nor the addresshash are included in the modified record, as they should if the record would be emplaced.

### PR DESCRIPTION
**Pull request by prcolaco**
fixes #29 

Please double check...

### CODE REVIEW

File: contracts/fio.system/src/voting.cpp [hunk 1]
```diff
@@ -1245,7 +1249,6 @@ namespace eosiosystem {
                               //and not having a proxy, its kind of a null vote, so dont emplace unless isproxy is true.
 
             uint64_t id = _voters.available_primary_key();
-            uint128_t addresshash = string_to_uint128_hash(fio_address.c_str());
```
Comment (dapixcasey) [inline contracts/fio.system/src/voting.cpp:1248]: This change is needed. Lets make a PR with just this change.

File: contracts/fio.system/src/voting.cpp [hunk 2]
```diff
@@ -1225,12 +1225,16 @@ namespace eosiosystem {
...
                 votersbyowner.modify(pitr, same_payer, [&](auto &p) {
+                    p.fioaddress = fio_address;
+                    p.addresshash = addresshash;
                     p.is_proxy = isproxy;
                     p.is_auto_proxy = false;
                 });
             }else if (!isproxy) { //this is how we undo/clear a proxy
                 name nm;
                 votersbyowner.modify(pitr, same_payer, [&](auto &p) {
+                    p.fioaddress = "";      // TODO: placed here to revert to same state as from regproducer, not sure if correct, verify...
```
Comment (dapixcasey) [inline contracts/fio.system/src/voting.cpp:1236]: Not needed. See Issue Notes.

File: contracts/fio.system/src/voting.cpp [hunk 3]
```diff
@@ -1225,12 +1225,16 @@ namespace eosiosystem {
...
+                    p.addresshash = 0;      // TODO: placed here to revert to same state as from regproducer, not sure if correct, verify...
```
Comment (dapixcasey) [inline contracts/fio.system/src/voting.cpp:1237]: Not needed. See Issue Notes.

File: contracts/fio.system/src/voting.cpp [hunk 4]
```diff
@@ -1225,12 +1225,16 @@ namespace eosiosystem {
             //check(!isproxy || !pitr->proxy, "account that uses a proxy is not allowed to become a proxy");
             if (isproxy && !pitr->proxy) {
                 votersbyowner.modify(pitr, same_payer, [&](auto &p) {
+                    p.fioaddress = fio_address;
+                    p.addresshash = addresshash;
```
Comment (dapixcasey) [inline contracts/fio.system/src/voting.cpp:1229]: Not needed. See Issue Notes.

File: contracts/fio.system/src/voting.cpp [hunk 5]
```diff
@@ -1225,12 +1225,16 @@ namespace eosiosystem {
             //check(!isproxy || !pitr->proxy, "account that uses a proxy is not allowed to become a proxy");
             if (isproxy && !pitr->proxy) {
                 votersbyowner.modify(pitr, same_payer, [&](auto &p) {
+                    p.fioaddress = fio_address;
```
Comment (dapixcasey) [inline contracts/fio.system/src/voting.cpp:1228]: Not needed. See Issue Notes.

**Review by dapixcasey — CHANGES_REQUESTED** [review]
Please remove the unneeded changes and leave the one accepted change.

File: contracts/fio.system/src/voting.cpp (hunk 2 above)
Comment (prcolaco) [reply contracts/fio.system/src/voting.cpp:1236]: Please check my reply in the issue. This is a workaround for the problem that happened with our proxy. Probably there's a better way to implement a definitive solution, but this solves the problem in the mid term.

File: contracts/fio.system/src/voting.cpp (hunk 3 above)
Comment (prcolaco) [reply contracts/fio.system/src/voting.cpp:1237]: Please check my reply in the issue. This is a workaround for the problem that happened with our proxy. Probably there's a better way to implement a definitive solution, but this solves the problem in the mid term.

File: contracts/fio.system/src/voting.cpp (hunk 4 above)
Comment (prcolaco) [reply contracts/fio.system/src/voting.cpp:1229]: Please check my reply in the issue. This is a workaround for the problem that happened with our proxy. Probably there's a better way to implement a definitive solution, but this solves the problem in the mid term.

File: contracts/fio.system/src/voting.cpp (hunk 5 above)
Comment (prcolaco) [reply contracts/fio.system/src/voting.cpp:1228]: Please check my reply in the issue. This is a workaround for the problem that happened with our proxy. Probably there's a better way to implement a definitive solution, but this solves the problem in the mid term.

**Review by fiocasey — APPROVED** [review]

**Comment by fiocasey** [comment]
I've tested the above and was able to duplicate the issue on my end. Thank you @prcolaco for the PR and Issue. Please join us on Discord and continue to contribute!

**Pull request merged**